frontend/packages/pipelines-plugin/src/utils/pipeline-filter-reducer.ts

```diff
@@ -61,5 +61,11 @@ export const pipelineResourceTypeFilter = (filters, pipelineResource): boolean =
 
 export const taskRunFilterReducer = (taskRun): string => {
   const status = pipelineRunStatus(taskRun);
+  if (status === 'Running') {
+    const condition = taskRun.status?.conditions?.find((c) => c.type === 'Succeeded');
+    if (condition.reason === 'ExceededNodeResources') {
```

Comment:
If taskRun.status could be empty (see optional check one line above), and the conditions could not include a type === 'Successeded', you must check here again if the condition is defined.

Reply:
There are also more reasons why a TaskRun is pending. Maybe you can update the `pipelineRunStatus` instead. I now found the source code of the `tkn` cli which converts the "Succeeded" condition into a "human readable text".

Since the tkn cli do this for PipelineRun and TaskRun in a similar way, I think we could do this also for PipelineRuns. Otherwise I liked that you tried to ensure with this local change to not break any PipelineRun related check.

See https://github.com/tektoncd/cli/blob/master/pkg/formatted/k8s.go#L55-L82

Wdyt? :)

Reply:
Thanks, I Initially avoided modifying the PipelineRun check because I wasn't sure if PipelineRun and TaskRun could have the same reasons. Updated now to handle all the cases.